dsc_lib/src/configure/depends_on.rs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use regex::Regex;
+
+use crate::configure::config_doc::Resource;
+use crate::configure::Configuration;
+use crate::DscError;
+
+pub fn get_resource_invocation_order(config: &Configuration) -> Result<Vec<Resource>, DscError> {
+    let mut order: Vec<Resource> = Vec::new();
+    let depends_on_regex = Regex::new(r"^\[resourceId\('(?<type>[a-zA-Z0-9\.]+/[a-zA-Z0-9]+)','(?<name>[a-zA-Z0-9 ]+)'\)]$")?;
```

Comment:
Wondering if it might be useful, organizationally, to define common regexes in one place and re-use them. Like defining the regex for:

- Resource type names
- Instance names
- The resourceID syntax
- Other custom syntaxes that we add later, like for looking up parameters or variables or outputs.

Reply:
Yes, it would make sense to have them in one place.  One limitation with Rust I hit is that I can't define a const regex (or string) and use it as an attribute value which only takes a literal.

Reply:
I'll leave it here for now unless I can figure out a way to use the same regex in the attribute and in code.  It doesn't seem like Rust macros solve this.  This regex is currently the only one used in code.